l0_sampling/sketch.cpp

```diff
@@ -1,29 +1,74 @@
 #include "../include/sketch.h"
 #include <cassert>
+#include <iostream>
 
-Sketch::Sketch(vec_t n, long seed, double num_bucket_factor):
-    seed(seed), n(n), num_bucket_factor(num_bucket_factor) {
-  const unsigned num_buckets = bucket_gen(n, num_bucket_factor);
-  const unsigned num_guesses = guess_gen(n);
-  bucket_a = std::vector<vec_t>(num_buckets * num_guesses);
-  bucket_c = std::vector<vec_hash_t>(num_buckets * num_guesses);
+double Sketch::num_bucket_factor = 0.5;
+vec_t Sketch::n;
+size_t Sketch::num_elems;
+size_t Sketch::num_buckets;
+size_t Sketch::num_guesses;
+
+Sketch::SketchUniquePtr Sketch::makeSketch(const Sketch &old) {
+  return makeSketch(old.seed);
+}
+
+
+/*
+ * Static functions for creating sketches without a provided memory location
+ * these are useful for unit tests
+ */
+Sketch::SketchUniquePtr Sketch::makeSketch(long seed) {
```

Comment:
If these are only used for unit tests, they should be in the testing/utility files, not here.

Reply:
They are only used for unit tests

Reply:
So they should go into a testing file as a helper function.

Reply:
Did that